tests/sweep_framework/sweep_utils/utils.py

```diff
@@ -18,6 +19,17 @@
 from models.utility_functions import torch_random
 
 
+def get_device_grid_size():
+    device_name = os.environ.get("ARCH_NAME", os.environ.get("TT_ARCH_NAME", "default")).lower()
```

Comment:
it is better to avoid reliance on ARCH_NAME
@blozano-tt is actively removing reliance of the codebase on this environment variable
I think arch can be instead fetched from the device, I am sure grid can be deduced/ asked from the device too.
I wonder if there are api available for that 
@cfjchu  any change you know?